spyder/widgets/ipythonconsole/client.py

```diff
@@ -203,6 +204,11 @@ def configure_shellwidget(self, give_focus=True):
         self.shellwidget.sig_show_syspath.connect(self.show_syspath)
         self.shellwidget.sig_show_env.connect(self.show_env)
 
+        # To identify if the kernel is from Spyder and
+        # attach it to other plugins
+        self.shellwidget.sig_spyder_kernel.connect(
+                lambda : self.sig_spyder_kernel.emit(self))
+

```

Comment:
There's no need for these changes in `client.py`. We can directly connect to the signal emitted by `shellwidget` in `ipythonconsole.py`.

So please remove the signal above and this code here.